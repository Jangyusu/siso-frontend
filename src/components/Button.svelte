<script lang="ts">
  let buttonText = 'Button';

  const handleClick = () => {
    buttonText = 'Button Clicked';
  };
</script>

<h1>Test button</h1>
<button on:click={handleClick}>{buttonText}</button>
