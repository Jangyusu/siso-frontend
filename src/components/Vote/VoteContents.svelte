<script lang="ts">
  import {voteSubjects} from '@/store';

  export let vote = async (type: string) => {
    await voteSubjects.postVote({id, type});
  };
  export let id = '';
  export let title = 'no-title';
  export let agree_description = '';
  export let disagree_description = '';
</script>

<div class="flex flex-col items-center justify-center flex-1 pt-4 pb-2">
  <span
    class="w-max px-3 py-[2px] h-[22px] text-center text-white text-xs mb-[6px] rounded-[20px] bg-[rgba(255,255,255,0.3)] border border-[rgba(85,85,85,0.3)]"
    >라이프</span
  >
  <h3 class="text-2xl text-white text-center mb-[17px] w-[320px]">{title}</h3>
  <button class="mb-3 vote-btn" on:click={() => vote('agree')}
    >{agree_description}</button
  >
  <button class="vote-btn" on:click={() => vote('disagress')}
    >{disagree_description}</button
  >
</div>

<style lang="scss">
  .vote-btn {
    @apply w-full h-[150px] flex justify-center items-center bg-[rgba(255,255,255,0.75)] shadow-[0_0_20px_4px_rgba(0,0,0,0.06)]
  rounded-[15px] text-lg text-[#222222] font-bold px-[29px];
  }
</style>
