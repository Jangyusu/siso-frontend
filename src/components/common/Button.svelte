<script lang="ts">
  export let onClick = () => {
    console.log('button clicked');
  };
  export let src = '/';
  export let alt = '';
  export let text: string | number = '';
</script>

<button on:click={onClick} class={$$props.class} style={$$props.style}>
  <img {src} {alt} />
  {text}
</button>

<style lang="scss">
  button {
    padding: 0;
    border: none;
    background: transparent;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 1px;
    font-size: 11px;
    color: $White;
    &.circle {
      border-radius: 100%;
    }
    &.size-lg {
      width: 48px;
      height: 48px;
    }
    &.size-md {
      width: 36px;
      height: 36px;
    }
    &.size-sm {
      width: 24px;
      height: 24px;
    }
    &.img-sm {
      img {
        width: 24px;
        height: 24px;
      }
    }
    &.img-md {
      img {
        width: 36px;
        height: 36px;
      }
    }
  }

  img {
    width: 100%;
  }
</style>
