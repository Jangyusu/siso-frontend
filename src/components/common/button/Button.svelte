<script lang="ts">
  export let onClick = () => {
    console.log('button clicked');
  };
  export let src = '/';
  export let alt = '';
  export let text: string | number = '';
  export let size = 'md';
  export let imgSize = 'md';
</script>

<button
  on:click={onClick}
  class={`${size === 'lg' ? 'w-12 h-12' : 'w-9 h-9'} `}
  style={$$props.style}
>
  <img class={`${imgSize === 'lg' ? 'w-9' : 'w-6'} w-full`} {src} {alt} />
  {text}
</button>

<style lang="postcss">
  button {
    @apply flex flex-col justify-center items-center p-0 bg-transparent text-xs text-white border-0 gap-[1px];
  }
</style>
