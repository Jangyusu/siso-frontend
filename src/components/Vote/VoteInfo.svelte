<script lang="ts">
  import Button from '@/components/Common/Button.svelte';
</script>

<div class="w-full flex justify-end mb-[25px]">
  <div class="flex">
    <Button
      src="/images/icon/icn_people.svg"
      alt="voteCount"
      text="9M"
      size="lg"
      imgSize="md"
    />
    <Button
      src="/images/icon/icn_comment_line.svg"
      alt="commentCount"
      text="999"
      size="lg"
      imgSize="md"
    />
    <Button
      src="/images/icon/icn_heart_line.svg"
      alt="likeCount"
      text="9.9K"
      size="lg"
      imgSize="md"
    />
  </div>
</div>
