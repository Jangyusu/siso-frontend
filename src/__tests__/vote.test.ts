import '@testing-library/jest-dom';
import {render} from '@testing-library/svelte';

import Button from '@/components/Common/Button.svelte';
import VoteContents from '@/components/Vote/VoteContents.svelte';
import VoteInfo from '@/components/Vote/VoteInfo.svelte';

describe('Vote', () => {
  test('should have props', () => {
    const text = 'button description';
    const {getByText} = render(Button, {props: {text}});
    expect(getByText(text)).toBeInTheDocument();
  });

  test('should have props', () => {
    const title = 'vote-title';
    const {getByText} = render(VoteContents, {props: {title}});
    expect(getByText(title)).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    const {container} = render(VoteInfo);
    expect(container).toMatchSnapshot();
  });
});
