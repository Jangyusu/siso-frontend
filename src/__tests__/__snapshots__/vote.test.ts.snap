// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vote should match snapshot 1`] = `
<body>
  <div>
    <div
      class="w-full flex justify-end mb-[25px]"
    >
      <div
        class="flex"
      >
        <button
          class="w-12 h-12  svelte-1o342qa"
        >
          <img
            alt="voteCount"
            class="w-6 w-full"
            src="/images/icon/icn_people.svg"
          />
           
          9M
        </button>
         
        <button
          class="w-12 h-12  svelte-1o342qa"
        >
          <img
            alt="commentCount"
            class="w-6 w-full"
            src="/images/icon/icn_comment_line.svg"
          />
           
          999
        </button>
         
        <button
          class="w-12 h-12  svelte-1o342qa"
        >
          <img
            alt="likeCount"
            class="w-6 w-full"
            src="/images/icon/icn_heart_line.svg"
          />
           
          9.9K
        </button>
      </div>
    </div>
  </div>
</body>
`;
